src/index.ts

```diff
@@ -74,253 +74,272 @@ const TYPED_MESSAGE_SCHEMA = {
+  useV4 = true,
+): Buffer {
+  return ethUtil.keccak(encodeData(primaryType, data, types, useV4));
+}
 
-  /**
-   * Hashes the type of an object
-   *
-   * @param {string} primaryType - Root type to hash
-   * @param {Object} types - Type definitions
-   * @returns {Buffer} - Hash of an object
-   */
-  hashType(primaryType: string, types: Record<string, unknown>): Buffer {
-    return ethUtil.keccak(this.encodeType(primaryType, types));
-  },
+/**
+ * Hashes the type of an object
+ *
+ * @param {string} primaryType - Root type to hash
+ * @param {Object} types - Type definitions
+ * @returns {Buffer} - Hash of an object
+ */
+function hashType(
+  primaryType: string,
+  types: Record<string, MessageTypeProperty[]>,
+): Buffer {
+  return ethUtil.keccak(encodeType(primaryType, types));
+}
 
-  /**
-   * Removes properties from a message object that are not defined per EIP-712
-   *
-   * @param {Object} data - typed message object
-   * @returns {Object} - typed message object with only allowed fields
-   */
-  sanitizeData<T extends MessageTypes>(
-    data: TypedData | TypedMessage<T>,
-  ): TypedMessage<T> {
-    const sanitizedData: Partial<TypedMessage<T>> = {};
-    for (const key in TYPED_MESSAGE_SCHEMA.properties) {
-      if (data[key]) {
-        sanitizedData[key] = data[key];
-      }
-    }
-    if ('types' in sanitizedData) {
-      sanitizedData.types = { EIP712Domain: [], ...sanitizedData.types };
+/**
+ * Removes properties from a message object that are not defined per EIP-712
+ *
+ * @param {Object} data - typed message object
+ * @returns {Object} - typed message object with only allowed fields
+ */
+function sanitizeData<T extends MessageTypes>(
+  data: TypedData | TypedMessage<T>,
+): TypedMessage<T> {
+  const sanitizedData: Partial<TypedMessage<T>> = {};
+  for (const key in TYPED_MESSAGE_SCHEMA.properties) {
+    if (data[key]) {
+      sanitizedData[key] = data[key];
     }
-    return sanitizedData as Required<TypedMessage<T>>;
-  },
+  }
+  if ('types' in sanitizedData) {
+    sanitizedData.types = { EIP712Domain: [], ...sanitizedData.types };
+  }
+  return sanitizedData as Required<TypedMessage<T>>;
+}
 
-  /**
-   * Signs a typed message as per EIP-712 and returns its keccak hash
-   *
-   * @param {Object} typedData - Types message data to hash as per eip-712
-   * @returns {Buffer} - keccak hash of the resulting signed message
-   */
-  eip712Hash<T extends MessageTypes>(
-    typedData: Partial<TypedData | TypedMessage<T>>,
-    useV4 = true,
-  ): Buffer {
-    const sanitizedData = this.sanitizeData(typedData);
-    const parts = [Buffer.from('1901', 'hex')];
+/**
+ * Signs a typed message as per EIP-712 and returns its keccak hash
+ *
+ * @param {Object} typedData - Types message data to hash as per eip-712
+ * @returns {Buffer} - keccak hash of the resulting signed message
+ */
+function eip712Hash<T extends MessageTypes>(
+  typedData: TypedData | TypedMessage<T>,
```

Comment:
This was never actually passed in `Partial<>` data. Also `sanitizeData` requires that the typed data be complete.